api/handlers.go

```diff
@@ -0,0 +1,276 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+
+	"github.com/nhatvu148/business-day-go/models"
+	tools "github.com/nhatvu148/business-day-go/tools"
+	"github.com/rs/zerolog/log"
+)
+
+type BusinessDayResult struct {
+	Result bool   `json:"result"`
+	Error  string `json:"error"`
+}
+
+type CustomError struct {
+	msg string
+}
+
+func (m *CustomError) Error() string {
+	return m.msg
+}
+
+type CustomHoliday struct {
+	Id       int64  `json:"id"`
+	Date     string `json:"date"`
+	Category string `json:"category"`
+}
+
+type IdResponse struct {
+	Id int64 `json:"id"`
+}
+
+func (app *Application) CustomHolidayHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
```

Comment:
There are constants you can use in the HTTP package for method
```suggestion
	case http.MethodGet:
```